batchlib/base.py

```diff
@@ -326,6 +333,36 @@ def check_scale_factors(scale_factors):
         if scale_factors[0] != 1:
             raise ValueError("First scale factor must be 1, got %i" % scale_factors[0])
 
+    @staticmethod
+    def check_format(formats, key):
+        # if we don't have a key, the format must be custom
+        if key is None:
+            if formats is not None and formats not in ('custom', ['custom'], ('custom')):
+                raise ValueError(f"Incompatible format {format} for non-present key")
+            return ['custom']
+
+        assert isinstance(key, (tuple, list))
```

Comment:
If the key was passed as a single string in init, it is converted to a list / tuple until here?

Reply:
Yes, it should be converted. I am just having this `assert` here to double check.
I will add a comment on this, because otherwise it's a bit confusing.
(In general, this is a common python thing to use asserts to just check if something with the inner logic goes wrong.)